samza-core/src/main/java/org/apache/samza/storage/blobstore/util/BlobStoreUtil.java

```diff
@@ -232,6 +288,15 @@ public CompletionStage<Void> deleteSnapshotIndexBlob(String snapshotIndexBlobId,
    * @return A future that completes when all the async downloads completes
    */
   public CompletableFuture<Void> restoreDir(File baseDir, DirIndex dirIndex, Metadata metadata) {
+    return restoreDir(baseDir, dirIndex, metadata, false);
+  }
+
+  /**
+   * Non-blocking restore of a {@link SnapshotIndex} to local store by downloading all the files and sub-dirs associated
+   * with this remote snapshot. getDeletedFiles flag sets whether to attempt a get for deletedFiles or not.
+   * @return A future that completes when all the async downloads completes
+   */
+  public CompletableFuture<Void> restoreDir(File baseDir, DirIndex dirIndex, Metadata metadata, Boolean getDeletedFiles) {
```

Comment:
Don't need 2 variants for the util method.

Reply:
Updated/removed. Thanks